### PR TITLE
taste.py: add docstring for Taste

### PR DESCRIPTION
**Pull request by proishan11**
Added parameter description for Taste

Closes https://github.com/coala/coala/issues/3879

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: coalib/bearlib/aspects/taste.py [hunk 1]
```diff
@@ -55,6 +55,14 @@ def __init__(self, description: str='', suggested_values: tuple=(),
...
         :class:`coalib.bearlib.languages.Language`.
 
+        :param description:      str     A string describing taste.
+        :param suggested_values  tuple   A tuple containing the list of
+                                         possible values for the taste.
+        :param default           bool    The default value of taste.
+        :param languages         tuple   A tuple containing list of
+                                         languages, for which taste is
+                                         defined.
+
```
Comment (pratyushprakash) [inline coalib/bearlib/aspects/taste.py]: This should be at the end of the docstring

Comment (pratyushprakash) [inline coalib/bearlib/aspects/taste.py]: Also I don't think the types for each parameter are required. They are already mentioned in the signature

**Comment by proishan11** [comment]
Hello @pratyushprakash !
I am kind of caught up on what do I have to fix in the shortlog comment, is it because of the `:` or is there something else I am messing up. I followed [this link](http://api.coala.io/en/latest/Developers/Git_Basics.html#adding-the-files-and-commiting) for the rules regarding commit messages.

Thanks! :smile:

File: coalib/bearlib/aspects/taste.py [hunk 2]
```diff
@@ -61,6 +61,12 @@ def __init__(self, description: str='', suggested_values: tuple=(),
         aspectclass definition.
 
         The value cast type is defined via indexing on class level.
+        :param description:         A string describing taste.
```
Comment (pratyushprakash) [inline coalib/bearlib/aspects/taste.py]: IMO this should be `Description of the taste`. `A string` is redundant.

File: coalib/bearlib/aspects/taste.py [hunk 3]
```diff
@@ -61,6 +61,12 @@ def __init__(self, description: str='', suggested_values: tuple=(),
...
+        :param suggested_values     A tuple containing the list of possible
+                                    values for the taste.
+        :param default              The default value of taste.
```
Comment (pratyushprakash) [inline coalib/bearlib/aspects/taste.py]: `the taste`

File: coalib/bearlib/aspects/taste.py [hunk 4]
```diff
@@ -61,6 +61,12 @@ def __init__(self, description: str='', suggested_values: tuple=(),
...
+        :param languages            A tuple containing list of languages, for
+                                    which taste is defined.
```
Comment (pratyushprakash) [inline coalib/bearlib/aspects/taste.py]: `the taste`

File: coalib/bearlib/aspects/taste.py [hunk 5]
```diff
@@ -61,6 +61,12 @@ def __init__(self, description: str='', suggested_values: tuple=(),
         aspectclass definition.
 
         The value cast type is defined via indexing on class level.
+        :param description:         Description of the taste.
+        :param suggested_values     A tuple containing the list of possible
+                                    values for the taste.
+        :param default              Default value of the taste.
+        :param languages            A tuple containing list of languages, for
+                                    which the taste is defined.
```
Comment (pratyushprakash) [inline coalib/bearlib/aspects/taste.py]: There should be a `:` after every param name. Also add a new line at L63

Comment (proishan11) [inline coalib/bearlib/aspects/taste.py]: Oh! Just forgot about that!
Thanks :smile:

File: coalib/bearlib/aspects/taste.py [hunk 6]
```diff
@@ -61,6 +61,12 @@ def __init__(self, description: str='', suggested_values: tuple=(),
         aspectclass definition.
 
         The value cast type is defined via indexing on class level.
+        :param description:         Description of the taste.
```
Comment (pratyushprakash) [inline coalib/bearlib/aspects/taste.py]: Newline at L63

**Comment by Nosferatul** [comment]
Make the changes and rebase,  marked wip.

**Comment by supergr35** [comment]
Please rebase and repush, it'll fix your appveyor failing.

**Comment by adhikasp** [comment]
ack b3154ca4d505053f0f33754d5d4925b3ab71646e

**Comment by SanketDG** [comment]
needs rebase

**Comment by adtac** [comment]
@cherry-pick-bot

**Comment by adtac** [comment]
Merged here https://github.com/coala/coala/pull/3994

**Pull request closed without merging**